app/src/main/java/com/example/mykotlincalculator21/mvp/model/CalculatorModel.kt

```diff
@@ -1,16 +1,59 @@
 package com.example.mykotlincalculator21.mvp.model
 
 import com.example.mykotlincalculator21.mvp.contract.CalculatorContract
-import com.example.mykotlincalculator21.utils.EMPTY_STRING
+import com.example.mykotlincalculator21.utils.*
 
 class CalculatorModel : CalculatorContract.CalculatorModelContract {
     private var firstOperand: String = EMPTY_STRING
+    private var secondOperand: String = EMPTY_STRING
+    private var operator: String = EMPTY_STRING
+    private var result: String = EMPTY_STRING
 
     override fun saveNumber(number: String) {
-        firstOperand = number
+        if (operator.isEmpty()) {
+            firstOperand += number
+        } else {
+            secondOperand += number
+        }
     }
 
-    override fun getResult(): String {
-        return firstOperand
+    override fun getPartialResult(): String {
+        return (firstOperand + operator + secondOperand)
     }
+
+    override fun getFullResult(): String {
+        when (operator) {
+            OPERATOR_SUM -> result =
+                ((java.lang.Double.parseDouble(firstOperand) + java.lang.Double.parseDouble(
```

Comment:
you can import java.lang.Double so here only call Double

Reply:
Done